src/index.ts

```diff
@@ -51,6 +52,14 @@ export type AssertionConsumerServiceT = (
 // logout express handler
 export type LogoutT = () => Promise<IResponsePermanentRedirect>;
 
+export type CallbackPayloadType = "REQUEST" | "RESPONSE";
+
+export type WithSpidCallbackT = (
+  sourceIp: string | null,
+  payload: string,
+  timestamp: string,
```

Comment:
it looke like this is more a date than a timestamp
```suggestion
  createdAt: UTCISODateFromString,
```